From c4f8c1238783078e24222e7ae8541b509d90f139 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:01:26 -0400
Subject: [usb] correct the usb_disconnect() comment about usb_bus_list_lock

Message-id: <1410887463-20674-15-git-send-email-dzickus@redhat.com>
Patchwork-id: 95102
O-Subject: [RHEL7 PATCH 014/591] USB: correct the usb_disconnect() comment about usb_bus_list_lock
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit db8f2aa358958fe5f82e5b9ffa6e0abbaa4fc236
Author: Bjorn Helgaas <bhelgaas@google.com>
Date:   Fri Sep 13 13:57:34 2013 -0600

    USB: correct the usb_disconnect() comment about usb_bus_list_lock

    usb_disconnect() no longer acquires usb_bus_list_lock, so update its
    comment to that effect.

    Signed-off-by: Bjorn Helgaas <bhelgaas@google.com>
    Signed-off-by: Alan Stern <stern@rowland.harvard.edu>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/core/hub.c b/drivers/usb/core/hub.c
index a0210f4..2159f82 100644
--- a/drivers/usb/core/hub.c
+++ b/drivers/usb/core/hub.c
@@ -2023,8 +2023,8 @@ static void hub_free_dev(struct usb_device *udev)
  * Something got disconnected. Get rid of it and all of its children.
  *
  * If *pdev is a normal device then the parent hub must already be locked.
- * If *pdev is a root hub then this routine will acquire the
- * usb_bus_list_lock on behalf of the caller.
+ * If *pdev is a root hub then the caller must hold the usb_bus_list_lock,
+ * which protects the set of root hubs as well as the list of buses.
  *
  * Only hub drivers (including virtual root hub drivers for host
  * controllers) should ever call this.
-- 
1.7.1